deps/rabbitmq_mqtt/src/rabbit_mqtt_processor.erl

```diff
@@ -1185,65 +1185,58 @@ get_vhost(UserBin, SslLogin, Port) ->
-            end
+                    VhostFromConfig = rabbit_mqtt_util:env(vhost),
+                    {plugin_configuration_or_default_vhost, {VhostFromConfig, UserBin}};
+                VHostFromPortMapping ->
+                    {port_to_vhost_mapping, {VHostFromPortMapping, UserBin}}
+            end;
+        VHostUser ->
+            {vhost_in_username, VHostUser}
     end.
 
 get_vhost_ssl(UserBin, SslLoginName, Port) ->
-    UserVirtualHostMapping = rabbit_runtime_parameters:value_global(
-                               mqtt_default_vhosts
-                              ),
-    case get_vhost_from_user_mapping(SslLoginName, UserVirtualHostMapping) of
+    case get_vhost_from_user_mapping(SslLoginName) of
         undefined ->
-            PortVirtualHostMapping = rabbit_runtime_parameters:value_global(
-                                       mqtt_port_to_vhost_mapping
-                                      ),
-            case get_vhost_from_port_mapping(Port, PortVirtualHostMapping) of
+            case get_vhost_from_port_mapping(Port) of
                 undefined ->
-                    {vhost_in_username_or_default, get_vhost_username(UserBin)};
+                    case get_vhost_username(UserBin) of
+                        undefined ->
+                            VhostFromConfig = rabbit_mqtt_util:env(vhost),
+                            {plugin_configuration_or_default_vhost, {VhostFromConfig, UserBin}};
+                        VHostUser ->
+                            {vhost_in_username, VHostUser}
+                    end;
                 VHostFromPortMapping ->
                     {port_to_vhost_mapping, {VHostFromPortMapping, UserBin}}
             end;
         VHostFromCertMapping ->
             {client_cert_to_vhost_mapping, {VHostFromCertMapping, UserBin}}
     end.
 
-vhost_in_username(UserBin) ->
-    case application:get_env(?APP_NAME, ignore_colons_in_username) of
-        {ok, true} -> false;
-        _ ->
-            %% split at the last colon, disallowing colons in username
-            case re:split(UserBin, ":(?!.*?:)") of
-                [_, _]      -> true;
-                [UserBin]   -> false;
-                []          -> false
-            end
-    end.
-
 get_vhost_username(UserBin) ->
-    Default = {rabbit_mqtt_util:env(vhost), UserBin},
     case application:get_env(?APP_NAME, ignore_colons_in_username) of
-        {ok, true} -> Default;
+        {ok, true} -> undefined;
         _ ->
             %% split at the last colon, disallowing colons in username
             case re:split(UserBin, ":(?!.*?:)") of
                 [Vhost, UserName] -> {Vhost,  UserName};
-                [UserBin]         -> Default;
-                []                -> Default
+                [UserBin]         -> undefined;
+                []                -> undefined
             end
     end.
 
+get_vhost_from_user_mapping(none) ->
+    undefined;
```

Comment:
Looks like this is dead code?

Reply:
yes dead code, originally wanted to merge `get_vhost_no_ssl` and `get_vhost_ssl`, it's a leftover from that. thanks for noticing, deleting